Dockerfile

```diff
@@ -0,0 +1,47 @@
+# syntax=docker/dockerfile1
+ARG VERSION=18.04
+FROM ubuntu:$VERSION
+ARG PYMATGEN_API_KEY
+ENV USERNAME dev
+ENV MP_API_KEY $PYMATGEN_API_KEY
+WORKDIR /home/$USERNAME
+
+COPY . ./piro/
+
+RUN apt-get update \
+ && apt-get -y update \
+ && apt-get install -y \
+    software-properties-common \
+    build-essential \
+    git \
+ && apt-get clean -qq
+
+RUN add-apt-repository ppa:deadsnakes/ppa
+
+RUN apt-get update \
+ && apt-get -y update \
+ && apt-get install -y \
+    python3.7 \
+    python3.7-dev \
+    python3-pip \
+ && apt-get clean -qq
+
+RUN python3.7 -m pip install --upgrade \
+    pip \
+    setuptools \
+    wheel
+
+RUN python3.7 -m pip install --upgrade \
```

Comment:
Do these deps need to be installed explicitly?  Shouldn't setup.py take care of them?  I'd expect these installations to be of the non-pinned versions.

Reply:
Ahh good point, let me go try it out

Reply:
I stripped out a lot of the installs in 06416bbc74f97689edded22fc09c012f35eee621 PTAL, I had to keep `numpy` in there, otherwise the `pymatgen` install fails.